### PR TITLE
Add dou tu extension

### PR DESCRIPTION
**Pull request by gaoyang**
## Description

An extension for easy retrieval of web emoji images while chatting.
In China, this is a very common and convenient feature.

## Screencast

dou-tu-1

dou-tu-2

dou-tu-3

dou-tu-4

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/dou-tu/src/services/clipboardService.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+      if (files.length >= 10) files.forEach((file) => fs.unlinkSync(`${tempDir}/${file}`));
+    }
+
+    const fileName = imageUrl.substring(imageUrl.lastIndexOf("/") + 1);
+    const response = await fetch(imageUrl);
+    const data = await response.arrayBuffer();
+    const filePath = `${tempDir}/${fileName}`;
+    await fs.writeFileSync(filePath, new Uint8Array(data), "binary");
+    proc.exec(`osascript -e 'set the clipboard to (read (POSIX file "${filePath}") as JPEG picture)'`);
+    showHUD("已复制到剪切板");
```
Comment (pernielsentikaer) [inline extensions/dou-tu/src/services/clipboardService.ts:21]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/package.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "dou-tu",
+  "title": "斗图",
```
Comment (pernielsentikaer) [inline extensions/dou-tu/package.json:4]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/package.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  "description": "An extension for easy retrieval of web emoji images while chatting.(上班时间斗个图🌟)",
```
Comment (pernielsentikaer) [inline extensions/dou-tu/package.json:5]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/package.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  "author": "gaoyang",
+  "categories": [
+    "Media",
+    "Fun"
+  ],
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "斗图",
```
Comment (pernielsentikaer) [inline extensions/dou-tu/package.json:16]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/package.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      "description": "An extension for easy retrieval of web emoji images while chatting.(上班时间斗个图🌟)",
```
Comment (pernielsentikaer) [inline extensions/dou-tu/package.json:17]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Dou Tu Extension for Raycast
+
+上班时间斗个图 😄
```
Comment (pernielsentikaer) [inline extensions/dou-tu/README.md:3]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

File: extensions/dou-tu/CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+# 斗图 Changelog
+
+## [Add Dou Tu Extension] - 2022-08-31
+- 支持快速搜索表情包
+- 支持点击复制到剪切板
```
Comment (pernielsentikaer) [inline extensions/dou-tu/CHANGELOG.md:5]: At the moment, Raycast doesn't support localization and only supports US English. Therefore, please avoid introducing your custom way to localize your extension.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪

After you change the language I'm able to check it out, at the moment when using the arrow keys it just closes Raycast and shows a message, not sure about what's the though about that is.

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by gaoyang** [comment]
Hi @pernielsentikaer . I have modified the issue.

Also, would you consider adding a default unchecked option to the Grid component?
Currently it is causing me some inconvenience.

**Comment by pernielsentikaer** [comment]
Hi @gaoyang 👋 

Why does it cause you issues when the first is selected?

**Comment by gaoyang** [comment]
@pernielsentikaer 
I want to trigger something to be done on click, because there is always a default selected item.
causing the automatically selected item to not be clicked to by the user.

**Comment by gaoyang** [comment]
For API suggestions I moved to
https://github.com/raycast/extensions/issues/2833

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
As stated in #2833 you have to use the [ActionPanel](https://developers.raycast.com/api-reference/user-interface/actions#api-reference) instead

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by pernielsentikaer** [comment]
Thanks for the update, @gaoyang; still, having the option for clicking is not the behavior the rest of Raycast is having; therefore, can we remove it? Are you okay with just having the first option (like the rest of Raycast rely's on)

**Comment by gaoyang** [comment]
I just want users to be able to find emojis easily and quickly when chatting.
Put more focus on chatting.
Please let my extension pass, thanks @pernielsentikaer ! 😁

**Comment by pernielsentikaer** [comment]
The grid is not meant to be used like that UX-wise.

There’s focus/selection, and there’s an action. When you start intermingling them, it leads to unexpected interaction for most users.

Another problem is that the `select by click` option would also trigger when you just move the selection by keyboard; that would be pretty unexpected.

**Comment by gaoyang** [comment]
@pernielsentikaer  Click operations have been removed.

**Pull request merged**